tests/components/konnected/test_init.py

```diff
@@ -0,0 +1,601 @@
+                },
+            )
+            is True
+        )
+
+    assert len(mock_int.mock_calls) == 3
+    p_hass, p_entry = mock_int.mock_calls[0][1]
+
+    assert p_hass is hass
+    assert p_entry is entry
+
+
+async def test_unload_entry(hass, mock_panel):
+    """Test being able to unload an entry."""
+    entry = MockConfigEntry(
+        domain=konnected.DOMAIN, data={konnected.CONF_ID: "aabbccddeeff"}
+    )
+    entry.add_to_hass(hass)
+
+    assert await async_setup_component(hass, konnected.DOMAIN, {}) is True
+    assert hass.data[konnected.DOMAIN]["devices"].get("aabbccddeeff") is not None
+    assert await konnected.async_unload_entry(hass, entry)
+    assert hass.data[konnected.DOMAIN]["devices"] == {}
+
+
+async def test_api(hass, aiohttp_client, mock_panel):
+    """Test callback view."""
+    await async_setup_component(hass, "http", {"http": {}})
+
+    device_config = config_flow.CONFIG_ENTRY_SCHEMA(
+        {
+            "host": "1.2.3.4",
+            "port": 1234,
+            "id": "112233445566",
+            "model": "Konnected Pro",
+            "access_token": "abcdefgh",
+            "default_options": config_flow.OPTIONS_SCHEMA({config_flow.CONF_IO: {}}),
+        }
+    )
+
+    device_options = config_flow.OPTIONS_SCHEMA(
+        {
+            "io": {
+                "1": "Binary Sensor",
+                "2": "Binary Sensor",
+                "3": "Binary Sensor",
+                "4": "Digital Sensor",
+                "5": "Digital Sensor",
+                "6": "Switchable Output",
+                "out": "Switchable Output",
+            },
+            "binary_sensors": [
+                {"zone": "1", "type": "door"},
+                {"zone": "2", "type": "window", "name": "winder", "inverse": True},
+                {"zone": "3", "type": "door"},
+            ],
+            "sensors": [
+                {"zone": "4", "type": "dht"},
+                {"zone": "5", "type": "ds18b20", "name": "temper"},
+            ],
+            "switches": [
+                {
+                    "zone": "out",
+                    "name": "switcher",
+                    "activation": "low",
+                    "momentary": 50,
+                    "pause": 100,
+                    "repeat": 4,
+                },
+                {"zone": "6"},
+            ],
+        }
+    )
+
+    entry = MockConfigEntry(
+        domain="konnected",
+        title="Konnected Alarm Panel",
+        data=device_config,
+        options=device_options,
+    )
+    entry.add_to_hass(hass)
+
+    assert (
+        await async_setup_component(
+            hass,
+            konnected.DOMAIN,
+            {konnected.DOMAIN: {konnected.CONF_ACCESS_TOKEN: "globaltoken"}},
+        )
+        is True
+    )
+
+    client = await aiohttp_client(hass.http.app)
+
+    # Test the get endpoint for switch status polling
+    resp = await client.get("/api/konnected")
+    assert resp.status == 404  # no device provided
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4986898)